lib/usb/usb_standard.c

```diff
@@ -337,13 +375,26 @@ static int usb_standard_get_interface(usbd_device *usbd_dev,
 				      uint8_t **buf, uint16_t *len)
 {
 	uint8_t *cur_altsetting;
-	const struct usb_config_descriptor *cfx = &usbd_dev->config[usbd_dev->current_config - 1];
+	const struct usb_config_descriptor *cfx;
+
+	if (! usbd_dev->current_config) {
+		return USBD_REQ_NOTSUPP;
+	}
+
+	/* FIXME: assuming (usbd_dev->current_config - 1) offset
+	 *   gives the configuration descriptor "desc" which has
+	 *   desc->bConfigurationValue == usbd_dev->current_config */
+	cfx = &usbd_dev->config[usbd_dev->current_config - 1];
 
 	if (req->wIndex >= cfx->bNumInterfaces) {
 		return USBD_REQ_NOTSUPP;
 	}
 
 	*len = 1;
+	/* FIXME: assuming (req->wIndex) offset
+	 *   gives the interface descriptor "iface" which has
+	 *   iface->bInterfaceNumber == req->wIndex and
+	 *     iface->bAlternateSetting == 0 */
```

Comment:
are these really fixmes? isn't this how it's kept?  it's not an assumption, they _are_ kept like that by the stack?